tests/e2e/flowtest/flowtest_helpers.py

```diff
@@ -0,0 +1,232 @@
+import logging
+
+from ocs_ci.ocs import node, platform_nodes, defaults, exceptions
+from ocs_ci.ocs.node import wait_for_nodes_status
+from ocs_ci.ocs.resources import pod as pod_helpers
+from ocs_ci.ocs.resources.pod import wait_for_storage_pods
+from ocs_ci.utility.utils import TimeoutSampler
+from tests.manage.mcg.helpers import s3_delete_object, s3_put_object, s3_get_object
+from tests.sanity_helpers import Sanity
+from ocs_ci.framework.testlib import ignore_leftovers
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.pvc import delete_pvcs
+from tests import helpers
+from tests.manage.z_cluster.pvc_ops import create_pvcs
+
+logger = logging.getLogger(__name__)
+
+
+class FlowOperations:
+    """
+    Flow based operations class
+
+    """
+    def __init__(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    def validate_cluster(self, node_status=False, pod_status=False, operation_name=""):
+        """
+        Validates various ceph and ocs cluster checks
+
+        Args:
+            node_status (bool): Verifies node is Ready
+            pod_status (bool): Verifies StorageCluster pods in expected state
+            operation_name (str): Name of the operation, to Tag
+
+        """
+        logger.info(f"{operation_name}: Verifying cluster health")
+        self.sanity_helpers.health_check(cluster_check=False, tries=100)
+        if node_status:
+            logger.info(f"{operation_name}: Verifying whether node is ready")
+            wait_for_nodes_status(status=constants.NODE_READY, timeout=300)
+        if pod_status:
+            logger.info(f"{operation_name}: Verifying StorageCluster pods are in running/completed state")
+            wait_for_storage_pods(timeout=300), 'Some pods were not in expected state'
+
+    def node_operations_entry_criteria(
+        self, node_type, number_of_nodes, operation_name="Node Operation", network_fail_time=None
+    ):
+        """
+        Entry criteria function for node related operations
+
+        Args:
+            node_type (str): Type of node
+            number_of_nodes (int): Number of nodes
+            operation_name (str): Name of the node operation
+            network_fail_time (int): Total time to fail the network in a node
+
+        Returns:
+            tuple: containing the params used in Node operations
+
+        """
+        self.validate_cluster(node_status=True, operation_name=operation_name)
+
+        logger.info(f"Getting parameters related to: {operation_name}")
+        typed_nodes = node.get_typed_nodes(
+            node_type=node_type, num_of_nodes=number_of_nodes
+        )
+        factory = platform_nodes.PlatformNodesFactory()
+        nodes = factory.get_nodes_platform()
```

Comment:
nodes instance is already created in confest.py, do we need to create it again?

Reply:
ack